Fix BoosterCatalogEndpointIT.shouldRespondWithBooster error because Catalog is not ready

```
[ERROR]   BoosterCatalogEndpointIT.shouldRespondWithBooster:75 1 expectation failed.
Expected status code <200> but was <404>.
```

If the catalog is not ready before the request is made, we get this error.

To fix, create a new endpoint to know if the catalog is ready.
Wait for the catalog to be ready to make the request.

--- a/web/src/main/java/io/fabric8/launcher/web/endpoints/BoosterCatalogEndpoint.java
+++ b/web/src/main/java/io/fabric8/launcher/web/endpoints/BoosterCatalogEndpoint.java
@@ -22,6 +22,7 @@ import io.fabric8.launcher.addon.BoosterCatalogFactory;
 import io.fabric8.launcher.booster.catalog.rhoar.Mission;
 import io.fabric8.launcher.booster.catalog.rhoar.RhoarBooster;
 import io.fabric8.launcher.booster.catalog.rhoar.RhoarBoosterCatalog;
+import io.fabric8.launcher.booster.catalog.rhoar.RhoarBoosterCatalogService;
 import io.fabric8.launcher.booster.catalog.rhoar.Runtime;
 import io.fabric8.launcher.booster.catalog.rhoar.Version;
 
@@ -139,4 +140,15 @@ public class BoosterCatalogEndpoint {
         boosterCatalogFactory.reset();
         return Response.ok().build();
     }
+
+    /**
+     * Used in integration tests
+     */
+    @GET
+    @Path("/wait")
+    public Response waitForIndex() throws Exception {
+        ((RhoarBoosterCatalogService) catalog).index().get();
+        return Response.ok().build();
+    }
+
 }

--- a/web/src/test/java/io/fabric8/launcher/web/endpoints/BoosterCatalogEndpointIT.java
+++ b/web/src/test/java/io/fabric8/launcher/web/endpoints/BoosterCatalogEndpointIT.java
@@ -24,6 +24,7 @@ import io.restassured.builder.RequestSpecBuilder;
 import io.restassured.specification.RequestSpecification;
 import org.jboss.arquillian.container.test.api.RunAsClient;
 import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
@@ -35,9 +36,20 @@ import static org.hamcrest.core.Is.isA;
 @RunAsClient
 public class BoosterCatalogEndpointIT extends BaseResourceIT {
 
-
     private RequestSpecification configureEndpoint() {
         return new RequestSpecBuilder().setBaseUri(UriBuilder.fromUri(deploymentUri).path("api").path("booster-catalog").build()).build();
+    }
+
+
+    @Before
+    public void waitUntilEndpointIsReady() {
+        given()
+                .spec(configureEndpoint())
+        .when()
+                .get("/wait")
+        .then()
+                .assertThat().statusCode(200);
+
     }
 
     @Test

